cmd/kubeadm/app/apis/kubeadm/v1alpha1/conversion_test.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1_test
+
+import (
+	"testing"
+
+	"k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	"k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1alpha1"
+)
+
+func TestUpgradeBootstrapTokens(t *testing.T) {
+	testcases := []struct {
+		name        string
+		in          *v1alpha1.MasterConfiguration
+		out         *kubeadm.MasterConfiguration
+		expectedOut *kubeadm.MasterConfiguration
+		expectError bool
+	}{
+		{
+			name: "empty configs should create at least one token",
+			in:   &v1alpha1.MasterConfiguration{},
+			out:  &kubeadm.MasterConfiguration{},
+			expectedOut: &kubeadm.MasterConfiguration{
+				BootstrapTokens: []kubeadm.BootstrapToken{
+					kubeadm.BootstrapToken{
+						Token: nil,
+					},
+				},
+			},
+			expectError: false,
+		},
+		{
```

Comment:
also add the case that `in.Token` is something valid

Reply:
Also add a test that makes sure TTL, Groups, Usages, etc. are preserved here as well

Reply:
done ✅